app/models/discovery_rule.rb

```diff
@@ -36,13 +36,31 @@ class DiscoveryRule < ApplicationRecord
     end
   }
 
+  def unique_priority
+      %w(locations organizations).each do |taxonomy|
+        discovery_rule_ids = TaxableTaxonomy.where(
+          taxable_type: 'DiscoveryRule',
+          taxonomy_id: send("#{taxonomy.singularize}_ids")).pluck(:taxable_id)
+        discovery_rules = DiscoveryRule.unscoped.where(id: discovery_rule_ids)
+        if discovery_rules.pluck(:priority).include?(priority)
+          errors.add(:priority, _("is already taken"))
+        end
```

Comment:
I think we should be able to do the check in a single query:

    DiscoveryRule.unscoped.where(id: discovery_rule_ids, priority: priority)

Can you also improve the error message and indicate which taxonomy (org or loc) is conflicting and present the name as well?